packages/block-editor/src/components/iframe/content.scss

```diff
@@ -32,8 +33,6 @@
 
 	body {
 		min-height: calc((#{$inner-height} - #{$total-frame-height}) / #{$scale});
-		display: flex;
-		flex-direction: column;
```

Comment:
Why was this removed? I believe this will break filling the page to fit the canvas height, cc @richtabor

Reply:
The context is in #65915. Either these rules persist on body, _both zoomed in and zoomed out_, or they are removed. Having them just in one case, causes the layout shift shown in 65915.

Reply:
I can reproduce it from the commit right before this, but _I think_ this didn't happen with zoom-out mode with TT4 just a few months ago. Maybe something changed that caused this glitchy behaviour.

![zoom-out-glitch](https://github.com/user-attachments/assets/e1d8a34a-c0a0-4724-b98d-5545421fcd3f)

The problem is that now the footer will no longer be pushed to the bottom, so we'll have to revert these lines in particular.

|Before|After|
|-|-|
|<img width="1464" alt="Screenshot 2024-10-16 at 17 24 47" src="https://github.com/user-attachments/assets/b565eee0-81ec-4296-a7f2-46a25ca18f69">|<img width="1464" alt="Screenshot 2024-10-16 at 17 22 38" src="https://github.com/user-attachments/assets/781749fe-260d-484d-ab69-eef54609c71e">|

Reply:
It's fine in the site editor, it's the _post_ editor that's the problem due to how the Title block has margins that collapse.

Reply:
> The problem is that now the footer will no longer be pushed to the bottom

I think it was an understood tradeoff. The flex display on the body was also the cause of #63519 so bringing it back brings that back as well.

Reply:
This is not a tradeoff 😃 this is removing a feature that I added half a year ago

Reply:
@ellatrix do we need the rule to apply at all times? Methinks that maybe when there is no template, there is no footer to be pushed down. So then apply flex only when there is a template?

Reply:
I didn’t mean to disparage anything by calling it a tradeoff, but fixing a couple things while knowingly breaking another seems to fit the definition to me. Though, yes, perhaps it doesn’t have to break.

Though I find it odd that for that feature we justify layout shift, since that’s what pushing the footer is. It would seem better if there were a way to orchestrate it after the zoom out transition and include some helpful indicator like expanding one of the insertion points.

Reply:
Clarifying further, theoretically it could also be the site editor. It's any editor where the first block uses a _top margin_. Any such margin will collapse when switching between flex and not block. That also means in the post editor, if you turn "Show template" on or off, you can switch between maybe having top margins or not:

![abstracted](https://github.com/user-attachments/assets/dcb36308-0f37-4324-8d15-53c9aa743353)

I'd be happy to restore the rule so it exists both zoomed and unzoomed. But the layout shift has to be fixed somehow.

Reply:
> I'd be happy to restore the rule so it exists both zoomed and unzoomed

Maybe I am missing something but to add a such a rule on the `body` (especially unzoomed) seems contrary to the isolation of UI and theme styling that the iframe is supposed to provide. Do we mandate that a block based themes do not control `body` styling? If so, I guess that’s what I'm missing. Otherwise, it seems like a non-starter—i.e. we should not add the rule (at least while unzoomed, but I have my doubts it’s proper even when zoomed).

Reply:
> The problem is that now the footer will no longer be pushed to the bottom

I’ve opened up #66388 to restore the feature in a different way that should be less liable to have other impacts. I already requested a review from some of the folks here but I'm mentioning it here in case anyone else like to have a look.